server/backend/api/app/crud/services/ext_app_user_search_service.py

```diff
@@ -0,0 +1,218 @@
+         "value":["00001011","00002011"]}]
+        """
+        roles_list = []  # hold list of roles for the user in ExtRoleWithScopeSchema
+        role_index = {}  # dict to look up existing role in roles_list
+        for xref in user.fam_user_role_xref:
+            role: FamRole = xref.role
+            if role.application_id != self.application_id:
+                continue
+
+            role_name = role.role_name
+            display_name = role.display_name
+            scope_type = None
+            current_scope_value = None
+            has_parent_role = role.parent_role is not None
+            if has_parent_role:
+                role_name = role.parent_role.role_name  # use parent's role name for child roles
+                display_name = role.parent_role.display_name  # use parent's display name for child roles
+                scope_type = ScopeType.FOREST_CLIENT  # FAM currently only has FOREST_CLIENT scope type
+                current_scope_value = role.forest_client_relation.forest_client_number
+
+            # If role_name already exists, merge scope_value
+            if role_name in role_index:
+                idx = role_index[role_name]
+                if current_scope_value not in roles_list[idx]["value"]:
+                    roles_list[idx]["value"].append(current_scope_value)
+            else:
+                # add new role entry
+                role_index[role_name] = len(roles_list)
+                roles_list.append({
+                    "applicationName": role.application.application_name,
+                    "roleName": role_name,
+                    "roleDisplayName": display_name,
+                    "scopeType": scope_type,
+                    "value": [current_scope_value] if current_scope_value else []
+                })
+        return roles_list
+
+    def _map_user_type_code_to_idp_type(self, user_type_code: str) -> IDPType:
+        """
+        Maps FamUser.user_type_code to IdpType enum for API response.
+        Example: 'I'(db) -> 'IDIR'(for API)
+        """
+        if user_type_code == UserType.IDIR:
+            return IDPType.IDIR
+        elif user_type_code == UserType.BCEID:
+            return IDPType.BCEID
+        else:
+            return IDPType.BCSC
+
+    def _apply_user_type_code_filter(self, user_role_stmt: Select, idp_type: IDPType) -> str:
+        """
+        Applies filtering to the statement based on the provided idp_type, maps to UserType for db filtering.
+        Example: 'IDIR'(for API) -> 'I'(db)  => where(FamUser.user_type_code == UserType.IDIR)
+        """
+        if idp_type is None:
+            return user_role_stmt
+
+        elif idp_type == IDPType.IDIR:
+            user_role_stmt = user_role_stmt.where(FamUser.user_type_code == UserType.IDIR)
+
+        elif idp_type == IDPType.BCEID:
+            user_role_stmt = user_role_stmt.where(FamUser.user_type_code == UserType.BCEID)
+
+        else:
```

Comment:
Should be explicit on this scenario - I assume idp_type == BCSC here? If we ever add a 4th IDP type, we'd want this logic to fail, not silently do the wrong thing.

Reply:
Thanks for bring that up Basil, yes, the parameter `idp_type` is an enum with 3 values only. But that's a good thought that I missed. I changed it for explicit condition.